docs/for.rst

```diff
@@ -43,3 +43,309 @@ in as a ``json`` parameter, converted to a set of ``json`` objects with
 Note that any optional properties values must be explicitly set to ``null``.
 They cannot be set to ``undefined`` or ommitted; doing so will cause a runtime
 error.
+
+Here's a more complex example, demonstrating how to complete a nested insert
+with conflicts on the inner items. First, take a look at the schema for this
+database:
+
+.. code-block:: sdl
+
+    module default {
+      type Character {
+        required property name -> str {
+          constraint exclusive;
+        }
+        property portrayed_by -> str;
+        multi link movies -> Movie;
+      }
+      type Movie {
+        required property title -> str {
+          constraint exclusive;
+        };
+        property release_year -> int64;
+      }
+    }
+
+Note that the ``Movie`` type's ``title`` property has an exclusive constraint.
+
+Here's the data we want to bulk insert:
+
+.. code-block:: js
+
+    [
+      {
+        portrayed_by: "Robert Downey Jr.",
+        name: "Iron Man",
+        movies: ["Iron Man", "Iron Man 2", "Iron Man 3"]
+      },
+      {
+        portrayed_by: "Chris Evans",
+        name: "Captain America",
+        movies: [
+          "Captain America: The First Avenger",
+          "The Avengers",
+          "Captain America: The Winter Soldier",
+        ]
+      },
+      {
+        portrayed_by: "Mark Ruffalo",
+        name: "The Hulk",
+        movies: ["The Avengers", "Iron Man 3", "Avengers: Age of Ultron"]
+      }
+    ]
+
+This is potentially a problem because some of the characters appear in the same
+movies. We can't just naively insert all the movies because we'll eventually
+hit a conflict. Since we're going to write this as a single query, chaining
+``.unlessConflict`` on our query won't help. It only handles conflicts with
+objects that existed *before* the current query.
+
+Let's look at a query that can accomplish this insert, and then we'll break it
+down.
+
+.. code-block:: typescript
+
+    const query = e.params(
+      {
+        characters: e.array(
+          e.tuple({
+            portrayed_by: e.str,
+            name: e.str,
+            movies: e.array(e.str),
+          })
+        ),
+      },
+      (params) => {
+        const movies = e.for(
+          e.op(
+            "distinct",
+            e.array_unpack(e.array_unpack(params.characters).movies)
+          ),
+          (movie) => {
```

Comment:
I would rename this variable to something else like `movieTitle`, so it's obviously different from `movie` in the `unlessConflict`.